spec/struct.dd

```diff
@@ -555,10 +622,161 @@ $(H2 $(LNAME2 disable_default_construction, Disabling Default Struct Constructio
             T t;          // error: default construction is disabled
         }
         ---
+        )
 
         $(BEST_PRACTICE Disabling default construction is useful when the default value,
         such as `null`, is not acceptable.)
 
+
+$(H2 $(LNAME2 UnionConstructor, Union Constructors))
+
+        $(P Unions are constructed in the same way as structs.)
+
+
+$(H2 $(LNAME2 field-init, Field initialization inside constructor))
+
+        $(P In a constructor body, if a delegate constructor is called,
+        all field assignments are considered assignments.
+        Otherwise, the first instance of field assignment is
+        its initialization, and assignments of the form `field = expression`
+        is treated as equivalent to `typeof(field)(expression)`.
+        The values of fields may be read before initialization or construction
+        with a delegate constructor.
```

Comment:
Except if the field is an aggregate declaration with default construction disabled. In that case, the compiler will (or should) issue an error.

Reply:
I'm not sure that change is right. In any case, again, here I am documenting what is, not what it should be.